### PR TITLE
Add CpuMpDebugProtocol for Debug Reporting AP Stack Buffers

### PR DESCRIPTION
**Pull request by TaylorBeebe**
## Description

Add CpuMpDebugProtocol for reporting AP stack buffers. This protocol was originally added to collect the AP stack info from the DXE paging audit.

https://github.com/microsoft/mu_plus/issues/102

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

Collecting the protocol from the DxePagingAuditTestApp

## Integration Instructions

N/A

### CODE REVIEW

**Review by kenlautner — APPROVED** [review]

File: UefiCpuPkg/Library/MpInitLib/DxeMpInitLib.inf [hunk 1]
```diff
@@ -62,6 +62,7 @@
 [Protocols]
   gEfiTimerArchProtocolGuid                     ## SOMETIMES_CONSUMES
   gEfiMemoryAttributeProtocolGuid               ## CONSUMES ## MU_CHANGE
+  gCpuMpDebugProtocolGuid                       ## CONSUMES ## MU_CHANGE
```
Comment (makubacki) [inline UefiCpuPkg/Library/MpInitLib/DxeMpInitLib.inf]: This library produces the protocol, right?

File: UefiCpuPkg/Library/MpInitLib/DxeMpLib.c [hunk 2]
```diff
@@ -159,6 +171,63 @@ BufferRemoveNoExecute (
...
+  UINTN  StackBuffer,
+  UINTN  StackSize,
+  UINTN  CpuNumber
+  )
+{
+  CPU_MP_DEBUG_PROTOCOL  *Entry;
+
+  Entry = AllocatePool (sizeof (CPU_MP_DEBUG_PROTOCOL));
+  if (Entry == NULL) {
+    DEBUG ((DEBUG_INFO, "%a - Failed to allocate memory!\n", __FUNCTION__));
```
Comment (makubacki) [inline UefiCpuPkg/Library/MpInitLib/DxeMpLib.c]:
```suggestion
    DEBUG ((DEBUG_ERROR, "%a - Failed to allocate memory!\n", __FUNCTION__));

```

File: UefiCpuPkg/UefiCpuPkg.dec [hunk 3]
```diff
@@ -91,6 +91,10 @@
   ## Include/Protocol/SmmExceptionTestProtocol.h
   gSmmExceptionTestProtocolGuid   = { 0xb76383a1, 0x0e70, 0x4a3f, { 0x86, 0xb4, 0xc6, 0x13, 0x4c, 0x8e, 0x57, 0x23 }}
 
+  # MU_CHANGE - Add protocol for reporting multi-processor debug info
+  #
+  ## Include/Protocol/MpDebug.h
```
Comment (makubacki) [inline UefiCpuPkg/UefiCpuPkg.dec]:
```suggestion
  ## Include/Protocol/CpuMpDebug.h

```

**Review by makubacki — APPROVED** [review]

File: UefiCpuPkg/Library/MpInitLib/DxeMpInitLib.inf (hunk 1 above)
Comment (TaylorBeebe) [reply UefiCpuPkg/Library/MpInitLib/DxeMpInitLib.inf]: Fixed

File: UefiCpuPkg/UefiCpuPkg.dec (hunk 3 above)
Comment (TaylorBeebe) [reply UefiCpuPkg/UefiCpuPkg.dec]: Fixed

**Pull request merged**